src/main/java/com/epam/izh/rd/online/repository/SimpleAuthorRepository.java

```diff
@@ -0,0 +1,69 @@
+package com.epam.izh.rd.online.repository;
+
+import com.epam.izh.rd.online.entity.Author;
+
+import java.util.Arrays;
+
+public class SimpleAuthorRepository implements AuthorRepository{
+    private Author[] authors = new Author[0];
+
+    @Override
+    public boolean save(Author author) {
+        if (findByFullName(author.getName(), author.getLastName()) != null) {
+            return false;
+        }
+        resize();
+        authors[authors.length - 1] = author;
+        return true;
+    }
+
+    @Override
+    public Author findByFullName(String name, String lastname) {
+        for (Author author: authors) {
+            if (author.getName().equals(name) && author.getLastName().equals(lastname)) {
+                return author;
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public boolean remove(Author author) {
+        for (int i = 0; i < authors.length; i++) {
+            if (authors[i].equals(author)) {
+                authors[i] = null;
+                resize();
+                return true;
+            }
+        }
+        return false;
+    }
+
+    //можно ли завязываться на null? Например если что-то не нашли, то вернуть null?
+    private void resize() {
```

Comment:
the name of this method isn't good because it doesn't tell us what happens inside. It isn't obvious that a call when the array contains null values will remove these values, but otherwise, it will add a null value to the end of the array.
Actually, this (impossible to find a good name for some method) is the red flag "you are trying to do smth unclear in this method, it should be refactored"
Here you've joined two independent flows of internal repository logic: "on insert and "on remove". It's worth splitting these flows to separate methods or inline them to save() and remove methods.

Reply:
Made changes and removed this method.